geo/src/lib.rs

```diff
@@ -81,6 +75,18 @@
 //!
 //! - **[`FrechetDistance`](algorithm::frechet_distance::FrechetDistance)**: Calculate the similarity between [`LineString`]s using the Fréchet distance
 //!
+//! ## Topology
+//!
+//! - **[`Contains`](algorithm::contains::Contains)**: Calculate if a geometry contains another
+//!   geometry
+//! - **[`CoordinatePosition`](algorithm::coordinate_position::CoordinatePosition)**: Calculate
+//!   the position of a coordinate relative to a geometry
+//! - **[`HasDimensions`](algorithm::dimensions::HasDimensions)**: Determine the dimensions of a geometry
+//! - **[`Intersects`](algorithm::intersects::Intersects)**: Calculate if a geometry intersects
+//!   another geometry
+//! - **[`line_intersection`](algorithm::line_intersection::line_intersection)**: Calculates the
```

Comment:
The reason it's capitalized like this is because `line_intersection` is a free function, not a trait. I should have called this out explicitly, because I went back and fourth on it.

My first thought was on one end of the spectrum: "eventually we want everything to intersect with everything else, so I should just introduce a super generic intersection trait" like:
```
trait Intersection<T> {
    type Output;
    fn intersection(&self, other: T) -> Self::Output
}

impl Intersection<Line<F>> for Line<F> {
    type Output = Option<LineIntersection<F>>
    fn intersection(&self, other: Line<F>) -> Self::Output {
        ...
    }
}
```

But since there aren't any near-term plans to introduce impls other than `Intersection<Line> for Line`, I thought suggesting that we support a generic concept of `Intersection` might just end up confusing people.

So then I thought maybe a more specific trait:
```
trait LineIntersection {
    fn line_intersection(&self, other: Line<T>) -> LineIntersection<T>;
}

impl LineIntersection<T> for Line<T> {
    fn line_intersection(&self, other: Line<T>) -> LineIntersection<T>;
        ...
    }
}
```

I'd be OK with this, but I think I would only ever want to have one thing (Line<T>) implement that trait... so it seems like a lot of machinery, and if we *did* eventually introduce a more generic `Intersection` trait, this trait would likely become wholly redundant.

So that's why I eventually settled on just a bare non-trait function, and why it's capitalized that way in the docs. It seemed like a stable way to expose the functionality that would be less likely to surprise people.

If we did eventually add intersections for more than just lines, I'd be in favor of adding a generic `Intersecion<T>` trait.

WDYT?